src/main/java/com/example/bulletinboard/service/impl/AdServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.example.bulletinboard.service.impl;
+
+import com.example.bulletinboard.dto.AdDto;
+import com.example.bulletinboard.dto.CreateOrUpdateAd;
+import com.example.bulletinboard.repository.AdRepo;
+import com.example.bulletinboard.service.AdMapper;
+import com.example.bulletinboard.service.AdService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class AdServiceImpl implements AdService {
+    private final AdRepo adRepo;
+    private final AdMapper adMapper;
+
+    @Override
+    public AdDto create(AdDto ad) {
+        return adMapper.toDto(adRepo.save(adMapper.toAd(ad)));
+    }
+
+    @Override
+    public Optional<AdDto> getById(Integer id) {
+        return Optional.ofNullable(adMapper.toDto(adRepo.findById(id).orElse(null)));
```

Comment:
```suggestion
        return adMapper.toDto(adRepo.findById(id).orElseThrow(AdNotFoundException::new));
```
лучше так - кидать своё исключение со статусом 404